geest/core/polygons_per_grid_cell.py

```diff
@@ -29,16 +40,52 @@ def raster_polygon_grid_score(self):
         :param input_polygons: Layer of point features to count within each grid cell.
         """
 
-        # Create grid
-        self.h_spacing = 100
-        self.v_spacing = 100
-        create_grid = GridCreator(h_spacing=self.h_spacing, v_spacing=self.v_spacing)
-        output_dir = os.path.join("output")
-        merged_output_path = os.path.join(output_dir, "merged_grid.shp")
-        grid_layer = create_grid.create_grids(
-            self.country_boundary, output_dir, self.crs, merged_output_path
+        output_dir = os.path.dirname(self.output_path)
+
+        # Define output directory and ensure it's created
+        os.makedirs(output_dir, exist_ok=True)
+
+        # Load grid layer from the Geopackage
+        geopackage_path = os.path.join(
+            self.working_dir, "study_area", "study_area.gpkg"
+        )
+        if not os.path.exists(geopackage_path):
+            raise ValueError(f"Geopackage not found at {geopackage_path}.")
+
+        grid_layer = QgsVectorLayer(
+            f"{geopackage_path}|layername=study_area_grid", "merged_grid", "ogr"
+        )
+
+        area_layer = QgsVectorLayer(
+            f"{geopackage_path}|layername=study_area_polygons",
+            "study_area_polygons",
+            "ogr",
+        )
+
+        geometries = [feature.geometry() for feature in area_layer.getFeatures()]
+
+        # Combine all geometries into one using unaryUnion
+        area_geometry = QgsGeometry.unaryUnion(geometries)
```

Comment:
Is this logic sequence needed? We already have the study_area_bbox and it should already be grid aligned (if not better to fix it in study_area.py)

Reply:
Also if we are doing the analysis island by island we only need the single poiygon of hte island and its bbox ...